File: ABC/351-400/374/c.cpp
#include <bits/stdc++.h>
using namespace std;
#define rep(i,n) for(int i=0;i<n;++i)
using ll = long long;
int main(){
    int n;
    cin >> n;
    vector<ll> k(n);
    rep(i,n) cin >> k[i];
    int n2 = 1<<n;
    ll ans = 1e18;
    rep(i,n2){
        ll a=0,b=0;
        rep(j,n){
            if(i >> j&1){
                a += k[j];
            }else{
                b += k[j];
            }
        }
        ll now = max(a,b);
        ans = min(ans,now);
    }
    cout << ans << endl;
    return 0;
}

File: ABC/351-400/374/c.py
def main():
  n = int(input())
  k = list(map(int,input().split()))
  ans = 1e18
  for i in range(1<<n):
    a,b = 0,0
    for j in range(n):
      if i>>j&1:
        a += k[j]
      else:
        b += k[j]
    ans = min(max(a,b),ans)
  print(ans)

if __name__ == '__main__':
  main()


